lib/FIR/mixin.h

```diff
@@ -25,6 +30,7 @@ namespace Fortran::FIR {
 
 inline constexpr bool has_size(std::size_t size) { return size > 0; }
 
+// implementation of a (moveable) sum type (variant)
 template<typename T, typename E = void> struct SumTypeMixin {};
```

Comment:
How does this compare to `UNION_CLASS_BOILERPLATE` in `parse-tree.h`. I.e. how would I choose which to use if I wanted to create a sum type?

Reply:
I've been considering replacing this with the code from `parse`, but not urgently.